src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -2,108 +2,1037 @@
+                    Abort();
+                    throw;
+                }
+                catch (Exception exc)
+                {
+                    _state = WebSocketState.Closed;
+                    Abort();
+                    throw new WebSocketException(SR.net_webstatus_ConnectFailure, exc);
+                }
             }
-        }
 
-        public WebSocketState State
-        {
-            get
+            public override Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType messageType, bool endOfMessage, CancellationToken cancellationToken)
             {
-                throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
+                try
+                {
+                    ClientWebSocket.ThrowIfInvalidState(_state, _disposed, s_validSendStates);
+                    ThrowIfOperationInProgress(_lastSendAsync);
+                }
+                catch (Exception e)
+                {
+                    return Task.FromException(e);
+                }
+
+                Task t = SendFrameAsync(buffer, _lastSendWasFragment ? MessageOpcode.Continuation : ToMessageOpcode(messageType), endOfMessage, cancellationToken);
+                _lastSendWasFragment = !endOfMessage;
+
+                return t.Status == TaskStatus.RanToCompletion ?
+                    t :
+                    (_lastSendAsync = WithAbortAsync<int>(t, AbortErrorType.OperationCanceledException, cancellationToken));
             }
-        }
 
-        public string SubProtocol
-        {
-            get
+            public override Task<WebSocketReceiveResult> ReceiveAsync(ArraySegment<byte> buffer, CancellationToken cancellationToken)
             {
-                throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
+                try
+                {
+                    ClientWebSocket.ThrowIfInvalidState(_state, _disposed, s_validReceiveStates);
+                    ThrowIfOperationInProgress(_lastReceiveAsync);
+                }
+                catch (Exception e)
+                {
+                    return Task.FromException<WebSocketReceiveResult>(e);
+                }
+
+                Task<WebSocketReceiveResult> t = ReceiveAsyncPrivate(buffer, cancellationToken);
+                return t.Status == TaskStatus.RanToCompletion ?
+                    t :
+                    (_lastReceiveAsync = WithAbortAsync<WebSocketReceiveResult>(t, AbortErrorType.ClosedOrAbortedWebSocketException, cancellationToken));
             }
-        }
-        #endregion
 
-        public static WebSocketHandle Create()
-        {
-            return default(WebSocketHandle);
-        }
+            public override Task CloseAsync(WebSocketCloseStatus closeStatus, string statusDescription, CancellationToken cancellationToken)
+            {
+                ClientWebSocket.ThrowIfInvalidState(_state, _disposed, s_validCloseStates);
```

Comment:
I'm fine either way, I just noticed "two of these things don't look like the others" :)